scripts/install_submitter.py

```
@@ -0,0 +1,619 @@
+                    logger.debug("Found existing VRED script (base64)")
+                    return script_base64.strip()
+
+            logger.debug("No python script key found in preferences.xml")
+            return ""
+
+        except (ET.ParseError, OSError) as e:
+            logger.debug("Failed to read VRED preferences.xml: %s", e)
+            return ""
+
+    def install(
+        self,
+        destination: Optional[Path] = None,
+        verbose: bool = False,
+        auto_configure: bool = True,
+    ) -> bool:
+        """Install VRED submitter with optional automatic configuration."""
+
+        if not self.is_windows:
+            logger.warning("VRED Pro is only supported on Windows. Installation skipped.")
+            return False
+
+        if verbose:
+            logging.basicConfig(level=logging.DEBUG, format="%(levelname)s: %(message)s")
+        else:
+            logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
+
+        # Determine installation directory
+        install_dir = destination or self.get_default_install_directory()
+        logger.info("Installing VRED submitter to: %s", install_dir)
+
+        # Get files to install
+        files = self.get_submitter_files()
+        total_files = sum(len(file_list) for file_list in files.values())
+
+        if total_files == 0:
+            logger.error("No submitter files found to install")
+            return False
+
+        logger.info("Found %d files to install", total_files)
+
+        # Create directory structure
+        directories = self.create_directory_structure(install_dir)
+
+        # Copy files
+        if not self.copy_files(files, directories):
+            return False
+
+        # Set DEADLINE_VRED_MODULES env var so VRED can find submitter modules
+        if not self.set_deadline_vred_modules_env_var(install_dir):
+            logger.warning("Environment variable setup failed, but installation can continue")
```

Why are we ok with this continuing? I think if we're intending to use this script for automated workflows, this would lead to submissions failing later on. I think it'd be better to fail fast early, no?

---

Makes sense! I've changed it to fail fast if environment variable setup fails.